Issue/6279 post list missing featured image

nbradbury opened the pull request:
Fixes #6279 - the problem was due to choosing a featured image from the post content when the image is still uploading. This resulted in choosing a local file (ex: `/storage/emulated/0/Download/IMG_q5m29m.jpg`) as a featured image, and the post adapter didn't support displaying device images.

This PR corrects this by supporting device images, and also fixes a few problems in the existing featured image code.

**Note:** As part of this PR, I added [this routine](https://github.com/wordpress-mobile/WordPress-Android/blob/d5d7ae968dc260d6421bf42a45f25ba8c8dd0f55/WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostsListAdapter.java#L328) to load resized bitmaps more efficiently than similar routines in [ImageUtils](https://github.com/wordpress-mobile/WordPress-Utils-Android/blob/develop/WordPressUtils/src/main/java/org/wordpress/android/util/ImageUtils.java#L598). At some point we'll want to move that routine to `ImageUtils`, but for now I'd like to keep it inside `PostListAdapter`.

daniloercoli commented on WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostsListAdapter.java:
```diff
@@ -309,6 +305,46 @@ public void onClick(View v) {
+            imgFeatured.setVisibility(View.VISIBLE);
+            imgFeatured.setImageUrl(photonUrl, WPNetworkImageView.ImageType.PHOTO);
+        } else {
+            Bitmap bmp = fastResizeBitmap(imageUrl, mPhotonWidth);
```
Any reason to not use `Bitmap bmp = ImageUtils.getWPImageSpanThumbnailFromFilePath(context, imageUrl, mPhotonWidth);`  that is well tested and used in the app?

The actual implementation doesn't take in consideration orientation of the pictures, resulting in pictures turned 90° on some devices, and doesn't show the "poster" picture for videos.

daniloercoli commented on WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostsListAdapter.java:
```diff
@@ -309,6 +305,46 @@ public void onClick(View v) {
+        if (imageUrl == null) {
+            imgFeatured.setVisibility(View.GONE);
+        } else if (imageUrl.startsWith("http")) {
+            String photonUrl =  ReaderUtils.getResizedImageUrl(imageUrl, mPhotonWidth, mPhotonHeight, mSite.isPrivate());
```
Instead of `mSite.isPrivate()` should we use `SiteUtils.isPhotonCapable(mSite)`?
Otherwise self hosted sites can get redirected to Photon.

nbradbury commented:
@daniloercoli You're right, I should've stuck with the existing image resize. My concern was that since the resizing occurs on the main thread I wanted it to be as fast as possible, and our existing code could be faster. But that's something we should address separately rather than in this PR.

I've updated this PR in 6541dd6 to use the existing method and also switched to using `isPhotonCapable`, as you suggested.

daniloercoli commented on WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostsListAdapter.java:
```diff
@@ -729,34 +751,23 @@ protected Boolean doInBackground(Void... nada) {
-                    imageUrl = scanner.getLargestImage();
                 } else {
-                    imageUrl = null;
+                    imageUrl = new ReaderImageScanner(post.getContent(), mSite.isPrivate()).getLargestImage();
```
Sorry, I missed this in the first pass. Should we use `SiteUtils.isPhotonCapable(mSite)` here?

nbradbury replied:
Fixed in 50f8a8c

daniloercoli commented:
:shipit:
